client/src/components/assessments/ondemand/OndemandAssessment.js

```diff
@@ -287,6 +285,9 @@ const OnDemandAssessment = ({
     console.error(
       `Recurrence type is not ${RECURRENCE_TYPE.ON_DEMAND}. Component will not be rendered!`
     )
+    return null
+  } else if (!filteredAssessmentConfig) {
+    return null
   } else {
```

Comment:
```suggestion
  else if (filteredAssessmentConfig) {
```

is returning `null` required?

Reply:
It's so that we can distinguish between: assessment shouldn't be shown at all (`null`), and: assessment is there, but empty (in which case it will be an Object `{ … }`).